backend/fms_core/resources/sample.py

```diff
@@ -135,106 +139,118 @@ def import_obj(self, obj, data, dry_run):
 
-            try:
-                container_parent = Container.objects.get(barcode=location_barcode)
-            except Container.DoesNotExist:
-                if location_barcode:
-                    # If a parent container barcode was specified, raise a
-                    # better error message detailing what went wrong.
-                    # Otherwise, we assume it was left blank on purpose.
-                    raise Container.DoesNotExist(f"Container with barcode {location_barcode} does not exist")
-                container_parent = None
-
-            container_data = dict(
-                kind=normalized_container_kind,
-                name=get_normalized_str(data, "Container Name"),
-                barcode=get_normalized_str(data, "Container Barcode"),
-                **(dict(location=container_parent) if container_parent else dict(location__isnull=True)),
-            )
-
-            normalized_coords = get_normalized_str(data, "Location Coord")
-
-            if normalized_container_kind in SAMPLE_CONTAINER_KINDS_WITH_COORDS:
-                # Case where container itself has a coordinate system; in this
-                # case the SAMPLE gets the coordinates (e.g. with a plate.)
-                obj.coordinates = normalized_coords
-            else:
-                # Case where the container gets coordinates within the parent
-                # (e.g. a tube in a rack).
-                container_data["coordinates"] = normalized_coords
-
-            # If needed, create a sample-holding container to store the current
-            # sample; or retrieve an existing one with the correct barcode.
-            # This will throw an error if the kind specified mismatches with an
-            # existing barcode record in the database, which serves as an
-            # ad-hoc additional validation step.
-
-            container, _ = Container.objects.get_or_create(**container_data)
-            obj.container = container
+        elif field.attribute == "sample_kind_name":
+            obj.sample_kind = SampleKind.objects.get(name=data["Sample Kind"])
 
-            return
+        elif field.attribute == "container_barcode":
+            normalized_container_kind = get_normalized_str(data, "Container Kind").lower()
+            if normalized_container_kind in SAMPLE_CONTAINER_KINDS:
+                # Oddly enough, Location Coord is contextual - when Container Kind
+                # is one with coordinates, this specifies the sample's location
+                # within the container itself. Otherwise, it specifies the location
+                # of the container within the parent container.
+                # TODO: Ideally this should be tweaked
+
+                location_barcode = get_normalized_str(data, "Location Barcode")
+
+                try:
+                    container_parent = Container.objects.get(barcode=location_barcode)
+                except Container.DoesNotExist:
+                    if location_barcode:
```

Comment:
the logic here seems a bit over complicated or not straightforward? I would get the location_barcode out of the try/except loop. If there is no parent container and it is ok, we should not even attempt to create a container with an empty string location barcode